tmt/steps/execute/internal.py

```diff
@@ -81,21 +81,27 @@ def log(key, value=None, color=None, shift=1, level=1):
             self.verbose(key, value, color, shift=2, level=3)
 
         # Execute the test, save the output and return code
+        timeout = ''
         start = time.time()
         try:
             stdout = guest.execute(
                 test.test, cwd=workdir, env=environment,
-                join=True, interactive=self.get('interactive'), log=log)
+                join=True, interactive=self.get('interactive'), log=log,
+                timeout=tmt.utils.duration_to_seconds(test.duration))
             test.returncode = 0
         except tmt.utils.RunError as error:
             stdout = error.stdout
             test.returncode = error.returncode
+            if test.returncode == tmt.utils.PROCESS_KILLED:
```

Comment:
Same here, what if the process gets killed, by OOM f.e..

Reply:
Fixed by using extra timeout return code.